code/datums/components/connect_range.dm

```diff
@@ -0,0 +1,106 @@
+/**
+ * This component behaves similar to connect_loc_behalf but for all turfs in range, hooking into a signal on each of them.
+ * Just like connect_loc_behalf, It can react to that signal on behalf of a seperate listener.
+ * Good for components, though it carries some overhead. Can't be an element as that may lead to bugs.
+ */
+/datum/component/connect_range
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+
+	/// An assoc list of signal -> procpath to register to the loc this object is on.
+	var/list/connections
+	var/atom/tracked
+
+	/// The component will hook into signals only on turfs not farther from tracked than this.
+	var/range
+	/// Whether the component works when the movable isn't directly located on a turf.
+	var/works_in_containers
+
+/datum/component/connect_range/Initialize(atom/tracked, list/connections, range, works_in_containers = TRUE)
+	if(!isatom(tracked) || isarea(tracked) || range < 0)
+		return COMPONENT_INCOMPATIBLE
+	src.connections = connections
+	src.tracked = tracked
+	src.range = range
+	src.works_in_containers = works_in_containers
+
+/datum/component/connect_range/InheritComponent(datum/component/component, original, atom/tracked, list/connections, range, works_in_containers)
+	// Not equivalent. Checks if they are not the same list via shallow comparison.
+	if(!compare_list(src.connections, connections))
+		return
+	if(src.tracked != tracked)
+		set_tracked(tracked)
+	if(src.range == range && src.works_in_containers == works_in_containers)
+		return
+	//Unregister the signals with the old settings.
+	unregister_signals(isturf(tracked) ? tracked : tracked.loc)
```

Comment:
Why this instead of get_turf()? do you not want it to work on things inside containers or something?

Reply:
`unregister_signals` unregisters connection signals from the turfs in range but also the moved signal, which is used to keep the connections updated if the tracked is inside another object, from all movables containing tracked... that if the tracked is a movable. `connect_range` is an oddball here because the tracked object could as well be a turf, and since turf.loc is basically an area and get_turf(area) doesn't work right I had to do this.